presto-spi/src/main/java/com/facebook/presto/spi/type/DoubleType.java

```diff
@@ -60,10 +60,9 @@ public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int
     }
 
     @Override
-    public int hash(Block block, int position)
+    public long hash(Block block, int position)
     {
-        long value = block.getLong(position, 0);
-        return (int) (value ^ (value >>> 32));
+        return block.getLong(position, 0);
```

Comment:
I don't think it needs to be hashed. It wasn't before. AFAIU real hashing with nice distribution is done later in the code when it's really needed.